plan/physical_plan_builder.go

```diff
@@ -43,44 +44,100 @@ const (
 // JoinConcurrency means the number of goroutines that participate in joining.
 var JoinConcurrency = 5
 
-func getRowCountByIndexRanges(sc *variable.StatementContext, table *statistics.Table, indexRanges []*IndexRange, indexInfo *model.IndexInfo) (uint64, error) {
+func getRowCountByIndexRanges(sc *variable.StatementContext, statstbl *statistics.Table, indexRanges []*IndexRange, indexInfo *model.IndexInfo, tableInfo *model.TableInfo) (uint64, error) {
+	var offset int
+	for i := range tableInfo.Indices {
+		if tableInfo.Indices[i].Name.L == indexInfo.Name.L {
+			offset = i
+			break
+		}
+	}
+	if len(statstbl.Indices[offset].Numbers) == 0 {
```

Comment:
Is it possible that offset is out of `Indices` range? At that time, we also need to use pseudo table.